src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/sentencing/SentencingService.kt

```diff
@@ -379,6 +382,59 @@ class SentencingService(
     }
   }
 
+  private fun refreshCourtOrder(
+    courtEvent: CourtEvent,
+    offenderNo: String,
+  ) {
+    courtEvent.courtEventCharges.find { charge -> charge.resultCode1?.resultRequiresACourtOrder() ?: false }?.let {
+      existingCourtOrder(courtEvent.offenderBooking, courtEvent) ?: let {
+        courtOrderRepository.save(
+          CourtOrder(
+            offenderBooking = courtEvent.offenderBooking,
+            courtCase = courtEvent.courtCase!!,
+            courtEvent = courtEvent,
+            orderType = "AUTO",
+            courtDate = courtEvent.eventDate,
+            issuingCourt = courtEvent.prison,
```

Comment:
so I guess courtEvent.prison is misnamed since this would be the court  Agency right?

Reply:
yes - that's on my todo list to change